pkg/internal/api/common.go

```diff
@@ -133,3 +136,28 @@ func (c *ClientImpl) doRequest(ctx context.Context, req *http.Request) ([]byte,
 
 	return body, err
 }
+
+func (c *ClientImpl) runQuery(ctx context.Context, serviceID string, sql string, args ...interface{}) ([]byte, error) {
+	qry, err := sqlbuilder.ClickHouse.Interpolate(sql, args)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Is there any reason we don't leverage ClickHouse query parameters (they are available in API as per documentation: https://clickhouse.com/docs/en/get-started/query-endpoints#query-parameters ) and do query interpolation on the client-side?

Reply:
I guess ClickHouse can miss query parameters parser support for some of the statements.

Reply:
Yes exactly. Not all fields are supported as parameters and thus I prefer having a uniform query handling on the client side rather than an hybrid approach where some params are handled by the server and some by the client